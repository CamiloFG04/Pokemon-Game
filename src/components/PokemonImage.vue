<template>
<div class="pokemon-container">
  <img  v-if="!showPokemon"
    class="hidden-pokemon"
    :src="imgSrc"
    alt="pokemon"
  />
  <img v-else
    class="fade-in"
    :src="imgSrc"
    alt="pokemon"
  />

</div>
</template>

<script>
export default {
  props:{
    pokemonId:{
      type:Number,
      required: true,
    },
    showPokemon:{
      type:Boolean,
      required:true,
      default: false
    }
  },
  computed:{
    imgSrc(){
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    }
  }
};
</script>

<style scoped>
/* Pokemon Picture */
.pokemon-container {
  height: 200px;
  width: 400px;
  margin: 0 auto;
  position: relative;
}
 
img {
  height: 200px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
    filter: grayscale(100%) brightness(0%);
}
</style>