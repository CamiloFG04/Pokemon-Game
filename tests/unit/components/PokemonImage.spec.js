import { shallowMount } from "@vue/test-utils";

import PokemonImage from "@/components/PokemonImage";

describe('PokemonImage Component', () => {

    test('must be match with snapshot', () => {
        const wrapper = shallowMount(PokemonImage,{
            props:{
                pokemonId:1,
                showPokemon:false
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    });
    test('must be show the hidden image and the pokemon 100', () => {
        const wrapper = shallowMount(PokemonImage,{
            props:{
                pokemonId:100,
                showPokemon:false
            }
        })
        const [img1,img2] = wrapper.findAll('img')
        expect(img1.exists()).toBeTruthy()
        expect(img2).toBeFalsy()
        expect(img1.classes('hidden-pokemon')).toBeTruthy()
        expect(img1.attributes().src).toContain('100.svg')

        
    });
    test('must be show the pokemon image if showPokemon is true', () => {
        const wrapper = shallowMount(PokemonImage,{
            props:{
                pokemonId:100,
                showPokemon:true
            }
        })
        const img1 = wrapper.find('img')
        expect(img1.exists()).toBeTruthy()

        expect(img1.classes('hidden-pokemon')).toBeFalsy()
        expect(img1.classes('fade-in')).toBeTruthy()
    });
});