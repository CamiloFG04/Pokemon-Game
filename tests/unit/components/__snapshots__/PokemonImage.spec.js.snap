// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImage Component must be match with snapshot 1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon"></div>`;
